pvanalytics/quality/data_shifts.py

```diff
@@ -0,0 +1,306 @@
+        Penalty value passed to the ruptures changepoint detection method.
+
+    Returns
+    -------
+    None.
+    """
+    # Check that the time series has a datetime index, and consists of numeric
+    # values
+    if not isinstance(time_series.index, pd.DatetimeIndex):
+        raise TypeError('Must be a Pandas series with a datetime index.')
+    # Check that the time series is sampled on a daily basis
+    if pd.infer_freq(time_series.index) != "D":
+        warnings.warn("Time series frequency not set. Setting frequency to "
+                      "daily, and resampling the daily sum value.")
+        time_series = time_series.resample('d').sum()
+    return
+
+
+def _erroneous_filter(time_series):
+    """
+    Remove any outliers from the time series.
+
+    Parameters
+    ----------
+    time_series : Pandas series with datetime index.
+        Daily time series of a PV data stream, which can include irradiance
+        and power data streams. This series represents the summed daily values
+        of the particular data stream.
+
+    Returns
+    -------
+    time_series: Pandas series, with a datetime index.
+        Time series, after filtering out outliers. This includes removal
+        of stale repeat readings, negative readings, and data greater than
+        the 99th percentile or less than the 1st percentile.
+    """
+    # Detect and mask stale data
+    stale_mask = gaps.stale_values_round(time_series, window=6,
+                                         decimals=3, mark='tail')
+    time_series = time_series[~stale_mask]
+    # Set in negative values to NaN
+    time_series.loc[time_series <= 0] = np.nan
+    # Remove the top 10% and bottom 10% data points, and keep everything
+    # else intact
+    time_series[(time_series <= time_series.quantile(.01)) |
+                (time_series >= time_series.quantile(.99))] = np.nan
+    time_series = time_series.drop_duplicates()
+    return time_series
+
+
+def _preprocess_data(time_series, remove_seasonality):
+    """
+    Pre-process the time series, including the following:
+        1. Min-max normalization of the time series.
+        2. Removing seasonality from the time series.
+
+    Parameters
+    ----------
+    time_series : Pandas series with datetime index.
+        Daily time series of a PV data stream, which can include irradiance
+        and power data streams. This series represents the summed daily values
+        of the particular data stream.
+
+    Returns
+    -------
+    Pandas series with a datetime index:
+        Time series, after data processing. This includes min-max
+        normalization, and, if the time series is in greater than 2 years in
+        length, seasonality removal.
+    """
+    # Convert the time series to a dataframe to do the pre-processing
+    column_name = time_series.name
+    if column_name is None:
+        column_name = "value"
+        time_series = time_series.rename(column_name)
+    df = time_series.to_frame()
```

Comment:
Because the input is a single Series, and the constructed series is returned, is the `name` construction only so that the returned `Series.name` has a value? If that's not important, then the constructed Series name is only internal here and could be anything.